### PR TITLE
Add an optional tasks section to the PR template

### PR DESCRIPTION
**Pull request by chmurph2**
### Description

Adds an Optional Tasks section to the pull request template.
These are common, but important optional tasks that I sometimes forget to include in my pull requests.

### Changes

* Adds a "Please include" to the notes section, to match convention
* Adds an optional tasks section to the PR template

### Screenshots

updates

### CODE REVIEW

**Review by avinash-vllbh — APPROVED** [review]
Nice!! ❤️ it!

**Comment by chmurph2** [comment]
Cool, thanks! 

I'll merge it early next week to potentially give others time to weigh in with any feedback.

**Comment by drinks** [comment]
Great idea. A fair number of these tasks seem to be geared toward library releases, but tagging the release after merge is missing. I realize it’s not a step that would get completed while a pr is in review, but for a new maintainer what do you think the best place is to remind folks that 1 tag == 1 version and any changes thereafter must increment the patch number?

**Comment by coding-red-panda** [comment]
I don't know, I'm usually deleting more from the template than actually filling it out...

**Comment by avinash-vllbh** [comment]
> I don't know, I'm usually deleting more from the template than actually filling it out...

I think for project repos it's best to have a local `pull_request_template.md` to confirm to respective teams expectations. This should be more of a catch all and helps with repos (usually client gems) to have a standard.

File: .github/pull_request_template.md
Comment (coding-red-panda) [inline .github/pull_request_template.md:43]: If we're gearing this for proper Gem support then the following step must be added:

```
- [ ] Create a release in GitHub
```

Or at the very least mentioned somewhere so people don't forget and muck up the changelog for a gem.

**Comment by chmurph2** [comment]
@drinks and @coding-bunny: I think this latest change addresses both of your concerns:

updates

**Review by lbeougher — APPROVED** [review]

**Comment by coding-red-panda** [comment]
I'm happy with that.
Now to teach people NOT to point their gems to master branches....

**Pull request merged**
